src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeployedToEnvironmentNotifierStep.java

```diff
@@ -186,21 +164,37 @@ private static FormValidation validateEnvironmentUrl(@CheckForNull String enviro
         public FormValidation doCheckEnvironmentName(@AncestorInPath Item context,
```

Comment:
I pushed all validation down to the descriptor where it belongs

Reply:
Yeah, I was a bit iffy on having the validation in the step itself, but I also wanted to be able to reuse the validation I had in the descriptor in the constructor so that it wasn't possible to create an invalid step

Reply:
You can always do this by calling `getDescriptor` in the SCM :)